core-web/libs/edit-content/src/lib/feature/edit-content/store/edit-content.store.ts

```diff
@@ -1,203 +1,307 @@
+             * it navigates to the content view with the updated contentlet and actions.
+             * In case of an error, it updates the state with an error message.
+             *
+             * @param options The options required to fire the workflow action.
+             */
+            fireWorkflowAction: rxMethod<DotFireActionOptions<{ [key: string]: string | object }>>(
+                pipe(
+                    tap(() => patchState(store, { status: ComponentStatus.SAVING })),
+                    switchMap((options) => {
+                        return workflowActionsFireService.fireTo(options).pipe(
+                            tap((contentlet) => {
+                                if (!contentlet.inode) {
+                                    router.navigate(['/c/content']);
+                                }
+                            }),
+                            switchMap((contentlet) => {
+                                return forkJoin({
+                                    actions: workflowActionService.getByInode(
+                                        contentlet.inode,
+                                        DotRenderMode.EDITING
+                                    ),
+                                    contentlet: of(contentlet)
                                 });
-                            }
-                        )
-                    );
-                })
-            );
-        }
-    );
-
-    /**
-     * Fire the workflow action and update the contentlet and actions
-     *
-     * @private
-     * @param {(DotFireActionOptions<{ [key: string]: string | object }>)} options
-     * @return {*}  {Observable<{
-     *         actions: DotCMSWorkflowAction[];
-     *         contentlet: DotCMSContentlet;
-     *     }>}
-     * @memberof DotEditContentStore
-     */
-    private fireWorkflowAction(
-        options: DotFireActionOptions<{ [key: string]: string | object }>
-    ): Observable<{
-        actions: DotCMSWorkflowAction[];
-        contentlet: DotCMSContentlet;
-    }> {
-        return this.WorkflowActionsFireService.fireTo(options).pipe(
-            tap((contentlet) => {
-                if (!contentlet.inode) {
-                    this.router.navigate(['/c/content']);
+                            }),
+                            tapResponse({
+                                next: ({ contentlet, actions }) => {
+                                    router.navigate(['/content', contentlet.inode], {
+                                        replaceUrl: true,
+                                        queryParamsHandling: 'preserve'
+                                    });
+
+                                    patchState(store, {
+                                        contentlet,
+                                        actions,
+                                        status: ComponentStatus.LOADED,
+                                        error: null
```

Comment:
@oidacra Strick check: Type 'null' is not assignable to type 'string | undefined'

Reply:
fixed